library/cohesity_clone_vm.py

```diff
@@ -0,0 +1,392 @@
+                task_name=module.params.get('name')
+            )
+            module.exit_json(**status)
+        else:
+            raise__cohesity_exception__handler(str(json.loads(ex.context.response.raw_body)), module)
+    except Exception as error:
+        raise__cohesity_exception__handler(error, module)
+
+
+def get_cohesity_client(module):
+    '''
+    function to get cohesity cohesity client
+    :param module: object that holds parameters passed to the module
+    :return:
+    '''
+    cluster_vip = module.params.get('cluster')
+    username = module.params.get('username')
+    password = module.params.get('password')
+    domain = 'LOCAL'
+    if "/" in username:
+        user_domain = username.split("/")
+        username = user_domain[1]
+        domain = user_domain[0]
+
+    cohesity_client = CohesityClient(cluster_vip=cluster_vip,
+                                     username=username,
+                                     password=password,
+                                     domain=domain)
+    return cohesity_client
+
+
+def main():
+    # => Load the default arguments including those specific to the Cohesity clone task.
+    argument_spec = cohesity_common_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type='str', required=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+            job_name=dict(type='str', required=True),
+            view_name=dict(type='str', required=True),
+            backup_timestamp=dict(type='str', default=''),
+            environment=dict(choices=['VMware'], default='VMware'),
+            vm_names=dict(type='list', required=True),
+            wait_for_job=dict(type='bool', default=True),
+            prefix=dict(type='str', default=""),
+            suffix=dict(type='str', default=""),
+            power_on=dict(type='bool', default=True),
+            network_connected=dict(type='bool', default=True),
+            wait_minutes=dict(type=int, default=30),
+            resource_pool=dict(type='str', required=True)
+        )
+    )
+
+    # => Create a new module object
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+    results = dict(
+        changed=False,
+        msg="Attempting to manage Cohesity Clone",
+        state=module.params.get('state')
+    )
+
+    cohesity_client = get_cohesity_client(module)
+    clone_exists, clone_details = get_clone_task(
+        cohesity_client, module, False)
+
+    if module.check_mode:
+        check_mode_results = dict(
+            changed=False,
+            msg="Check Mode: Cohesity clone task doesn't exist",
+            id=""
+        )
+        if module.params.get('state') == "present":
+            if clone_exists:
+                check_mode_results[
+                    'msg'] = "Check Mode: Cohesity clone task is already present. No changes"
```

Comment:
can you please format this ? key[val] in same sentence

Reply:
I ran a autopep8 tool to fix pep8 errors, it formatted the code,  ok will change